src/mongocrypt-kek.c

```diff
@@ -196,6 +226,14 @@ _mongocrypt_kek_append (const _mongocrypt_kek_t *kek,
          BSON_APPEND_UTF8 (
             bson, "endpoint", kek->provider.gcp.endpoint->host_and_port);
       }
+   } else if (kek->kms_provider == MONGOCRYPT_KMS_PROVIDER_KMIP) {
+      BSON_APPEND_UTF8 (bson, "provider", "kmip");
+      if (kek->provider.kmip.endpoint) {
+         BSON_APPEND_UTF8 (bson, "endpoint", kek->provider.kmip.endpoint->host_and_port);
+      }
+      if (kek->provider.kmip.key_id) {
+         BSON_APPEND_UTF8 (bson, "keyId", kek->provider.kmip.key_id);
+      }
```

Comment:
Just to be sure, having no keyId is not an error?

Reply:
Good question. I think it should be an error in the case that `keyId` is not present in `_mongocrypt_kmip_kek_t` when `_mongocrypt_kek_append` is called.

Updated. The tests in `kek-tests.json` assume a successful parse implies append succeeds. I added a separate `expect_append` field.